repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/readme/ReadmePreProcessor.java

```diff
@@ -0,0 +1,96 @@
+    private final AtlasGraph graph;
+
+    public ReadmePreProcessor(AtlasTypeRegistry typeRegistry, EntityGraphRetriever entityRetriever, AtlasGraph graph) {
+        this.entityRetriever = entityRetriever;
+        this.typeRegistry = typeRegistry;
+        this.graph = graph;
+    }
+
+    @Override
+    public void processAttributes(AtlasStruct entityStruct, EntityMutationContext context,
+                                  EntityMutations.EntityOperation operation) throws AtlasBaseException {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("GlossaryPreProcessor.processAttributes: pre processing {}, {}", entityStruct.getAttribute(QUALIFIED_NAME), operation);
+        }
+
+        AtlasEntity entity = (AtlasEntity) entityStruct;
+
+        switch (operation) {
+            case CREATE:
+                processCreateReadme(entity, context);
+                break;
+            case UPDATE:
+                processUpdateReadme(entity, context.getVertex(entity.getGuid()));
+                break;
+        }
+    }
+
+    private void processCreateReadme(AtlasEntity entity, EntityMutationContext context) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("processCreateReadme");
+        String entityQualifiedName = createQualifiedName(entity);
+
+        entity.setAttribute(QUALIFIED_NAME, entityQualifiedName);
+
+        AtlasEntityType entityType = typeRegistry.getEntityTypeByName(entity.getTypeName());
+        AtlasVertex vertex = AtlasGraphUtilsV2.findByUniqueAttributes(this.graph, entityType, entity.getAttributes());
+        if(vertex != null && !vertex.getProperty(STATE_PROPERTY_KEY, String.class).equals(DELETED.name())){
+            String guidOfExistingReadme = GraphHelper.getGuid(vertex);
+            entity.setGuid(guidOfExistingReadme);
+            context.cacheEntity(guidOfExistingReadme, vertex, entityType);
+        }
+        else if(vertex != null){
+            vertex.setProperty(STATE_PROPERTY_KEY,ACTIVE.name());
+            context.cacheEntity(entity.getGuid(), vertex, entityType);
+        }
+
+        RequestContext.get().endMetricRecord(metricRecorder);
+    }
+    private void processUpdateReadme(AtlasStruct entity, AtlasVertex vertex) {
+
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("processUpdateReadme");
+        String vertexQnName = vertex.getProperty(QUALIFIED_NAME, String.class);
+        if(vertex.getProperty(STATE_PROPERTY_KEY, String.class).equals(DELETED.name())){
+            vertex.setProperty(STATE_PROPERTY_KEY,ACTIVE.name());
+        }
+
+        entity.setAttribute(QUALIFIED_NAME, vertexQnName);
+        RequestContext.get().endMetricRecord(metricRecorder);
+    }
+
+    public static String createQualifiedName(AtlasEntity entity) {
+        AtlasObjectId asset = (AtlasObjectId) entity.getRelationshipAttribute("asset");
+        return asset.getGuid() + "/readme" ;
```

Comment:
Though asset is mandatory here, add a null check if GUID is null here. What if user sends `uniqueAttributes` instead of GUID?